src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -399,6 +399,9 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             var connectors = obj["Connectors"].ToObject<IEnumerable<ConnectorModel>>(serializer);
 
             var info = new WorkspaceInfo(guid.ToString(), name, description, Dynamo.Models.RunType.Automatic);
+            //https://jira.autodesk.com/browse/QNTM-3872 
+            //Category should be set explictly from custom node workspace.
+            info.Category = obj["Category"].Value<string>();
```

Comment:
why is this necessary? - shouldn't the node workspaceInfo contain this?

Reply:
nope..Category is not in the constructor..in XML, this is done explicitly. `FromJSONDocument` does it correctly. again, in `serializtionConverters` this is set to different value.

Reply:
Same question, I think the cate is included in workspace info, you can see how customNodeWorkspace got initialized at https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Workspaces/CustomNodeWorkspaceModel.cs#L90

Reply:
This is after deserialization.

Reply:
hmm, but you can see above that workspaceInfo is not passed a category... I think Ram is right... wow.

Reply:
Wait how could **cate** be deserialized correctly before this change?

Reply:
yeah, I am wondering the same thing... Ram is saying it was only working with xml custom nodes but I am not sure about that...

Reply:
It is not actually. `Open the xml dyf - save - open - save : category will be null`

Reply:
if thats true... it is bad.

Reply:
you can test on the master branch..but I did the testing, and ran into this.

Reply:
I think Andreas ran into the same issue....